### DISCUSSION TITLE
How to properly access request param types in handlers where params are documented in parent router?

### DISCUSSION THREAD

**Discussion by cainenielsen** [honojs/hono#2299] [General]
I am pretty new to Hono so I am assuming I am missing something obvious, but I have exhausted the content on [hono.dev](hono.dev) trying to understand this so here we go.

My goal is to follow the recommendation of using nested `Hono()` instances for grouping routes and to support custom parameters in paths; eg. `/:someId`, which seems to work for handlers set directly on hono instances with `basePath()` but not on handlers set on child instances using `.route()`. I assume the context of the request parameter and its type is losted between the two instances.

My questions are as follows...
- Is this a bug that needs to be solved?
- Could this be a feature that should be requested?
- Is there a method, type, or overload that I can use to fix this?
- Is there a better or more common approach to building with Hono that covers this situation and circumvents this issue?

Example: 

```ts
// parentRouter.ts

import childRouter from './childRouter'

const parentRouter = new Hono().basePath('/:someId')

parentRouter.get('/some-path', async (context) => {
  const someId = context.req.param('someId') // -> ✔️ `const someId: string`
  return context.text(someId)
})

parentRouter.route('/child', childRouter)
```

```ts
// childrouter.ts

const childrouter = new Hono()

childrouter.get('/some-path', async (context) => {
  const someId = context.req.param('someId') // -> ❌ `Argument of type 'string' is not assignable to parameter of type 'never'.ts(2345)`
  return context.text(someId)
})

export default childRouter
```

Thanks for your time and support.

**Comment by cainenielsen** [comment]
Ah, I just found a question I believe is similar.
https://github.com/orgs/honojs/discussions/2257

**Comment by yusukebe** [comment]
Hi @cainenielsen 

It is not a bug, nor one we do not plan to address at this time right now. However, it is good to leave it as a feature request while it will be difficult to implement.

The workaround is this:

```ts
childrouter.get('/some-path', async (context) => {
  const someId = context.req.param<'/:someId'>('someId')
  return context.text(someId)
})
```
